src/main/java/seedu/address/commons/events/ui/CommandInputChangedEvent.java

```diff
@@ -0,0 +1,21 @@
+package seedu.address.commons.events.ui;
+
+import seedu.address.commons.events.BaseEvent;
+
+/**
+ * Indicates a change of input in command box
+ * Contains the full string of input in the command box
+ */
+public class CommandInputChangedEvent extends BaseEvent {
+
+    public final String currentInput;
```

Comment:
Is `public` the best access modifier for `currentInput`? As a rule of thumb start all data members as private, and depending on their usage promote them to protected, and if absolutely necessary, to public.